content/blog/2020/08/2020-08-28-github-checks-api-plugin-coding-phase-3.adoc

```diff
@@ -0,0 +1,83 @@
+
+.Coding Phase 3 Demo
+video::S-pEgT3lzpk[youtube,width=800,height=420]
+
+=== Pipeline Support
+
+The pipeline support allows users to directly publish checks in their pipeline script without depending on any other consumers.
+
+image::/images/post-images/2020/08-github-checks-api-plugin-coding-phase-3/pipeline-check.png[Pipeline Checks]
+
+The check in the above screenshot is published by script:
+
+[source, groovy]
+----
+publishChecks name: 'pipeline check', title: 'pipeline ', summary: '# A pipeline check example', text: "## This check is published through the pipeline script", detailsURL: 'https://ci.jenkins.io'
+----
+
+_If you want to publish checks to GitHub, please install the link:https://github.com/jenkinsci/github-checks-plugin[GitHub implementation] and refer to the link:https://docs.github.com/en/rest/reference/checks[document] for the requirements of each field. A default value (build link) for `detailsURL` will be provided._
+
+This feature can be useful when many stages exist in your pipeline script and each takes a long time: you can publish a check for each stage to keep track of the build.
+
+=== Rerun Request Support
+
+The rerun request allows GitHub users to rerun the failed builds. When a build failed (which leads to a failed check), a `Re-run` button will be added automatically by GitHub.
+
+image::/images/post-images/2020/08-github-checks-api-plugin-coding-phase-3/failed-checks.png[Failed Checks]
+
+By clicking the `Re-run` button, Jenkins will reschedule a build for the last commit of this branch.
+
+_Since all checks of a commit are produced by a single build, you don't have to rerun all failed checks, just rerun any one of the failed check will refresh all checks._
+
+=== Git SCM Support
+
+Thanks to link:https://github.com/uhafner[Ullrich]'s great help, the plugin:github-checks[GitHub Checks Plugin] now supports link:https://github.com/jenkinsci/git-plugin[Git SCM].
+This means now you can publish checks for your freestyle project or any other projects that use Git SCM.
+
+=== Document
+
+link:https://github.com/jenkinsci/checks-api-plugin/blob/master/docs/consumers-guide.md[Consumers Guide] and link:https://github.com/jenkinsci/checks-api-plugin/blob/master/docs/implementation-guide.md[Implementation Guide] are now available.
+As a Jenkins developer, you can now start consuming our API or even providing an implementation for other SCM platforms beside GitHub.
+
+=== Acknowledgment
+
+The whole link:/projects/gsoc/2020/projects/github-checks/[GitHub Checks API project] is started as a Google Summer of Code project. Much appreciate my mentors (link:https://github.com/timja[Tim] and link:https://github.com/uhafner[Ullrich]) for their great help during the whole summer. Also huge thanks to the link:https://www.jenkins.io/sigs/gsoc/[Jenkins GSoC SIG] and the whole community for the technique support and resources.
```

Comment:
Shouldn't we use links to the bio in Jenkins.io?

Reply:
I don't know how to use bio links, I look at some other blogs, they all use github ones.

Reply:
I believe you could link to the bio in this format `link:blog/authors/<author-name>/`